modules/oe_theme_helper/src/Plugin/Block/SiteNavigationBlock.php

```diff
@@ -0,0 +1,196 @@
+    unset($options[0]);
+
+    $form['level'] = [
+      '#type' => 'select',
+      '#title' => $this->t('Initial visibility level'),
+      '#default_value' => $config['level'],
+      '#options' => $options,
+      '#description' => $this->t('The menu is only visible if the menu item for the current page is at this level or below it. Use level 1 to always display this menu.'),
+      '#required' => TRUE,
+    ];
+    return $form;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function blockSubmit($form, FormStateInterface $form_state) {
+    $this->configuration['level'] = $form_state->getValue('level');
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function build() {
+    $menu_name = $this->getDerivativeId();
+    $parameters = $this->menuTree->getCurrentRouteMenuTreeParameters($menu_name);
+
+    // Adjust the menu tree parameters based on the block's configuration.
+    $level = $this->configuration['level'];
+    $parameters->setMinDepth($level);
+    $parameters->setMaxDepth(min($level + 1, $this->menuTree->maxDepth()));
+
+    $site_info = $this->configFactory->get('system.site');
+    $build = [
+      '#theme' => 'oe_theme_helper_site_navigation',
+      '#site_name' => $site_info->get('name'),
+      '#menu_items' => [],
+    ];
+    $cacheable_metadata = CacheableMetadata::createFromRenderArray($build);
+    $cacheable_metadata->addCacheableDependency($site_info);
+
+    // If menu is empty we only print out the site name.
+    $tree = $this->menuTree->load($menu_name, $parameters);
+    if (empty($tree)) {
+      return $build;
+    }
+
+    // Build site tree and process its links.
+    $menu_build = $this->buildRenderableMenu($tree);
+    $cacheable_metadata->merge(CacheableMetadata::createFromRenderArray($menu_build));
+    $build['#menu_items'] = $this->getEclLinks($menu_build['#items']);
+    $cacheable_metadata->applyTo($build);
+
+    return $build;
+  }
+
+  /**
+   * Build menu renderable array.
+   *
+   * @param \Drupal\Core\Menu\MenuLinkTreeElement[] $tree
+   *   The menu tree, as returned from MenuLinkTreeInterface::load().
+   *
+   * @return array
+   *   A renderable array.
+   */
+  protected function buildRenderableMenu(array $tree): array {
+    $manipulators = [
+      ['callable' => 'menu.default_tree_manipulators:checkAccess'],
+      ['callable' => 'menu.default_tree_manipulators:generateIndexAndSort'],
+    ];
+    $tree = $this->menuTree->transform($tree, $manipulators);
+
+    return $this->menuTree->build($tree);
+  }
+
+  /**
+   * Massage data to be compliant with ECL navigation menu data structure.
+   *
+   * For each menu tree item we rename the following properties:
+   *
+   * - "title" into "label"
+   * - "url" into "href"
+   * - "in_active_trail" into "is_current"
+   *
+   * @param array $items
+   *   Menu tree renderable array.
+   *
+   * @return array
+   *   Array of ECL-compatible links.
+   */
+  protected function getEclLinks(array $items): array {
```

Comment:
Since we are handling the transformation here, we no longer need to do the same on the a preprocess: oe_theme_preprocess_menu__main

Reply:
Mh not really, that one stays as that is for system menu blocks, while this one is for our custom navigation block.